src/main/java/com/redhat/devtools/lsp4ij/features/rename/LSPRenameRefactoringDialog.java

```diff
@@ -174,8 +199,19 @@ static void doRename(@NotNull LSPRenameParams renameParams,
                 LSPRenameHandler.showErrorHint(editor, LanguageServerBundle.message("lsp.refactor.rename.cannot.be.renamed.error"));
             } else {
                 // Apply the rename from the LSP WorkspaceEdit list
-                WriteCommandAction
-                        .runWriteCommandAction(psiFile.getProject(), () -> workspaceEdits.forEach(workspaceEditData -> LSPIJUtils.applyWorkspaceEdit(workspaceEditData.edit())));
+                WriteCommandAction.runWriteCommandAction(psiFile.getProject(), () -> {
+                    workspaceEdits.forEach(workspaceEditData -> LSPIJUtils.applyWorkspaceEdit(workspaceEditData.edit()));
+
+                    // Update any found external references with the new name
+                    externalReferences.forEach(externalReference -> {
+                        // Don't let a single failed external reference keep us from updating other references
+                        try {

```

Comment:
For relational integrity purposes, it's important that failure of a single rename propagation not short-circuit the rest.